src/wp-includes/formatting.php

```diff
@@ -3161,7 +3155,42 @@ function wp_rel_callback( $matches, $rel ) {
 		}
 		$text = trim( $html );
 	}
-	return "<a $text rel=\"" . esc_attr( $rel ) . '">';
+
+	$rel_attr = $rel ? ' rel="' . esc_attr( $rel ) . '"' : '';
+
+	return "<a {$text}{$rel_attr}>";
+}
+
+/**
+ * Helper function used to determine if a URL is an internal URL.
+ *
+ * @since [version]
+ *
+ * @param string $href The URL to test.
+ * @return bool Returns true for internal URLs and false for all other URLs.
+ */
+function _wp_is_url_internal( $href ) {
+
+	if ( in_array( strtolower( wp_parse_url( $href, PHP_URL_SCHEME ) ), array( 'http', 'https' ), true ) ) {
+		/**
+		 * Filters the array of URL hosts which are considered internal.
+		 *
+		 * @since [version]
+		 *
+		 * @param array $internal_hosts An array of internal URL hostnames.
+		 */
+		$internal_hosts = apply_filters(
+			'wp_rel_callback_internal_hosts',
```

Comment:
I don't love this filter name, but also not sure what would be better. Maybe just `wp_rel_internal_hosts`?

Reply:
If moving to a new "core" utility function, I think `wp_internal_host_urls` might be more helpful. Maybe it should even be moved to it's own dedicated function? `wp_internal_host_urls()` which returns the filtered list only? I like writing code like this myself but didn't want to add additional new functions unless it seemed appropriate.